v2/dtos/requests/intervalaction.go

```diff
@@ -26,7 +26,14 @@ type AddIntervalActionRequest struct {
 // Validate satisfies the Validator interface
 func (request AddIntervalActionRequest) Validate() error {
 	err := v2.Validate(request)
-	return err
+	if err != nil {
+		return errors.NewCommonEdgeXWrapper(err)
+	}
+	err = request.Action.Address.Validate()
```

Comment:
The validator lib can validate the `Address ` without `dive`. We just need to verify the REST or MQTT protocol properties according to the different Address `type`.

Reply:
doesn't this handle verify the REST or MQTT validation w/o calling Address.Validate()?
```
validate:"oneof='REST' 'MQTT'"`
````

Reply:
Yes. The validator lib can verify the deeper struct without calling Address.Validate().
```
func (request AddIntervalActionRequest) Validate() error {
	err := v2.Validate(request)       // verify the intervalAction and address
	return err
	if err != nil {
		return errors.NewCommonEdgeXWrapper(err)
	}
	err = request.Action.Address.Validate()   // verify the address and the REST or MQTT protocol properties
```

Reply:
so why do we need
```
err = request.Action.Address.Validate()   // verify the address and the REST or MQTT protocol properties
```
if this does all the validation?
```
	err := v2.Validate(request)       // verify the intervalAction and address
```

Reply:
> err := v2.Validate(request)       // verify the intervalAction and address

Because it only verify the struct according to the validate tag. It doesn't call the Address.Validate() function to verify the protocol properties.

https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/dtos/address.go#L30-L44

Reply:
OK, I finally got it! ;-)